File: backend/Api/src/WorkoutApp.Application/DTOs/Workout/Workout/WorkoutResponse.cs
using WorkoutApp.Domain.Enums;

namespace WorkoutApp.Application.DTOs.Workout;

public sealed record WorkoutResponse(
    Guid Id,
    ExerciseType ExerciseType,
    int DurationMinutes,
    int CaloriesBurned,
    int Difficulty,
    int Fatigue,
    string? Notes,
    DateTime PerformedAt);

File: frontend/src/app/shared/dto/responses/workouts/workout-response.ts
import { ExerciseType } from '../../../models/exercise-type';

export interface WorkoutResponse {
  id: string;
  exerciseType: ExerciseType;
  durationMinutes: number;
  caloriesBurned: number;
  difficulty: number;
  fatigue: number;
  notes?: string;
  performedAt: string;
}